File: backend/components/metiers/administration_publique/tests/plugins/plugin_manager.test.py
# plugin_manager.test.py – Tests unitaires avancés pour PluginManager (Python)
import pytest
from .plugin_manager import PluginManager

class DummyPlugin:
    def run(self, x):
        return x * 2

def test_register_and_get_plugin():
    pm = PluginManager()
    plugin = DummyPlugin()
    pm.register('dummy', plugin)
    assert pm.get('dummy') is plugin

def test_list_plugins():
    pm = PluginManager()
    pm.register('a', DummyPlugin())
    pm.register('b', DummyPlugin())
    assert set(pm.list_plugins()) == {'a', 'b'}

def test_execute_plugin():
    pm = PluginManager()
    plugin = DummyPlugin()
    pm.register('dummy', plugin)
    assert pm.execute('dummy', 3) == 6

def test_unregister_plugin():
    pm = PluginManager()
    plugin = DummyPlugin()
    pm.register('dummy', plugin)
    pm.unregister('dummy')
    assert pm.get('dummy') is None

def test_execute_invalid_plugin():
    pm = PluginManager()
    with pytest.raises(ValueError):
        pm.execute('notfound', 1)

File: backend/components/metiers/administration_publique/tests/plugins/plugin_manager.test.js
// plugin_manager.test.js – Tests unitaires avancés pour PluginManager (JS)
class DummyPlugin {
  run(x) { return x * 2; }
}

const PluginManager = require('./plugin_manager');

describe('PluginManager', () => {
  it('enregistre et récupère un plugin', () => {
    const pm = new PluginManager();
    const plugin = new DummyPlugin();
    pm.register('dummy', plugin);
    expect(pm.get('dummy')).toBe(plugin);
  });

  it('liste les plugins', () => {
    const pm = new PluginManager();
    pm.register('a', new DummyPlugin());
    pm.register('b', new DummyPlugin());
    expect(pm.listPlugins().sort()).toEqual(['a', 'b']);
  });

  it('exécute un plugin', () => {
    const pm = new PluginManager();
    const plugin = new DummyPlugin();
    pm.register('dummy', plugin);
    expect(pm.execute('dummy', 3)).toBe(6);
  });

  it('désenregistre un plugin', () => {
    const pm = new PluginManager();
    const plugin = new DummyPlugin();
    pm.register('dummy', plugin);
    pm.unregister('dummy');
    expect(pm.get('dummy')).toBeUndefined();
  });

  it('lance une erreur si plugin non trouvé', () => {
    const pm = new PluginManager();
    expect(() => pm.execute('notfound', 1)).toThrow();
  });
});
